services/service.js

```diff
@@ -8,30 +8,27 @@ mongoose.connect('mongodb://test:test@ds257627.mlab.com:57627/first-blog-mern',
 
 class service {
 
+    constructor()
+    {
+        this.Data = {};
+    }
+
+    
+    getOnePost(req) {
+        return Post.find({_id: req.params.post_id});
+    };
+
+
     getAllPosts() {
-        let info = 'hello';
-        //promise
-        let transferData = function() {
-            return new Promise((resolve, reject) => {
-
-                Post.find({}, function(err, data) {
-                    if (err) throw (err);
-                    info = data;
-                    console.log(data);
-                });
-                resolve();
-            });
-        };
-        transferData().then(() => {
-            return info;
-        });
+        return Post.find({});
     };
-};
 
+    postNewPost(req) {
```

Comment:
`postNewPost` should not take in `req`, but some other params representing a new post

ie: a object that has fields `title`, `author`, and `body`, or 3 params representing `title`, `author`, and `body`

Reply:
fixed